### PR TITLE
feat(clone): added .Clone methods to all proto objects

### PR DESCRIPTION
**Pull request by gunnar-solo**
fix(code-generator): code generation now copies all *.go files (#453)

related https://github.com/solo-io/gloo/issues/5620

### CODE REVIEW

File: pkg/api/v1/resources/resource_interface.go [hunk 1]
```diff
@@ -44,16 +45,37 @@ func ProtoCast(res Resource) (ProtoResource, error) {
...
 	Resource
+	// Deprecated: prefer GetNamespacedStatuses()
 	GetStatus() *core.Status
+	// Deprecated: prefer SetNamespacedStatuses()
 	SetStatus(status *core.Status)
+	GetNamespacedStatuses() *core.NamespacedStatuses
+	SetNamespacedStatuses(namespacedStatuses *core.NamespacedStatuses)
+}
+
+type StatusGetter interface {
```
Comment (sam-heilbron) [inline pkg/api/v1/resources/resource_interface.go:56]: Why are these changes included?

**Review by sam-heilbron — CHANGES_REQUESTED** [review]
There are a few changes included in this PR that weren't included in the original. How did you backport/cherry-pick this work?

File: pkg/api/v1/resources/resource_interface.go (hunk 1 above)
Comment (gunnar-solo) [reply pkg/api/v1/resources/resource_interface.go:56]: > how did you backport...
Lazily, apparently.  Got hit with build issues with a basic cherry-pick, so I ended up reproducing the steps for the original work as a new commit
* update needed dependencies with `go get`
* replace all instances of `option (extproto.equal_all) = true;` with `option (extproto.equal_all) = true;\noption (extproto.clone_all) = true;`
* update `resource_interface.go` with a clean copy from `master` 
* update `pkg/code-generator/cmd/main.go` with a clean copy from `master`

**Review by sam-heilbron — APPROVED** [review]
